File: backend/src/main/scala/microservices/attraction/objects/ListAttractionsPlannerRequest.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ListAttractionsPlannerRequest(city: Option[String], keyword: Option[String], useDate: Option[String])
object ListAttractionsPlannerRequest:
  given sourceEncoder: Encoder[ListAttractionsPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[ListAttractionsPlannerRequest] = deriveDecoder

File: frontend/src/microservices/attraction/objects/ListAttractionsPlannerRequest.ts
// 本文件定义 attraction 模块的 `ListAttractionsPlannerRequest`，作为列表查询参数并提供 JSON 编解码。

export type ListAttractionsPlannerRequest = {
  city?: string
  keyword?: string
  useDate?: string
}

export const listAttractionsPlannerRequestFromJson = (json: string): ListAttractionsPlannerRequest =>
  JSON.parse(json) as ListAttractionsPlannerRequest

export const listAttractionsPlannerRequestToJson = (value: ListAttractionsPlannerRequest): string =>
  JSON.stringify(value)
